README.md

```diff
@@ -1774,6 +1773,44 @@ Disabling Auto-Init means that you _must_ explicitly call `terragrunt init` prio
 
 If Auto-Init is disabled, and terragrunt detects that `terraform init` needs to be called, then terragrunt will fail.
 
+### Auto-Retry
+
+_Auto-Retry_ is a feature of `terragrunt` that will automatically address situations where a `terraform` command needs to be re-run. 
+
+Terraform can fail with transient errors which can be addressed by simply retrying the command again. In the event `terragrunt` finds one of these errors, the command will be re-run again automatically.
+ 
+In the event of a module or provider update, `terraform` commands such as `plan` or `apply` will fail with a message to run `terraform init`. In this case `terragrunt` will re-run the `init` command and then retry the command. 
+
+**Example 1**
+
+```
+$ terragrunt plan
+...
+Error: Error loading modules: module consul: not found, may need to run 'terraform init'
+```
+
+In this case, as terraform knows it needs to re-run `init` auto-retry will rerun `init` and then call `plan` again.
+
+**Example 2**
+
+```
+$ terragrunt apply
+...
+Initializing provider plugins...
+- Checking for available provider plugins on https://releases.hashicorp.com...
+Error installing provider "template": error fetching checksums: Get https://releases.hashicorp.com/terraform-provider-template/1.0.0/terraform-provider-template_1.0.0_SHA256SUMS: net/http: TLS handshake timeout.
+```
+
+Terragrunt sees this error, and knows it is a transient error that can addressed by re-running the `apply` command.
+
+`auto-retry` will try a maximum of three times to re-run the command, at which point it will deem the error as not transient, and accept the terraform failure. Retries will occur when the error is encountered, pausing for 5 seconds between retries.
+
+Known errors that `auto-retry` will rerun, are maintained in the `TerragruntOptions.RetryableErrors` map. Future upgrades to `terragrunt` may include the ability to configure `auto-retry` - pass in additional error strings, configure max retries and set retry interval - via the `terragrunt` config.
+
+To disable `auto-retry`, use the `--terragrunt-no-auto-retry` command line option or set the `TERRAGRUNT_AUTO_RETRY` environment variable to `false`.
+
+If [Auto-Init](#auto-init) is disabled, `terragrunt` will not attempt to re-run `terraform init`.
```

Comment:
I would assume we don't want run init no matter what if auto-init is disabled. Is this a correct assumption?

Reply:
Yes, I think that's correct

Reply:
Hi @autero1 @brikis98 can I get sample on how to use the auto-retry function? Am a noobie using Terragrunt fro the first time.

Thanks!